internal/caching/cache_serverkeys.go

```diff
@@ -16,22 +15,21 @@ const (
 // ServerKeyCache contains the subset of functions needed for
 // a server key cache.
 type ServerKeyCache interface {
-	GetServerKey(request gomatrixserverlib.PublicKeyLookupRequest) (response gomatrixserverlib.PublicKeyLookupResult, ok bool)
+	GetServerKey(request gomatrixserverlib.PublicKeyLookupRequest, timestamp gomatrixserverlib.Timestamp) (response gomatrixserverlib.PublicKeyLookupResult, ok bool)
```

Comment:
It is not entirely clear what `timestamp` means - the current time? Can we opt out of checks by passing `0`? Please document.

Reply:
Also, please document the logic that implementations of this interface need to make e.g what happens if things are not found / found but expired / etc.

Reply:
Updated the comments here.